### PR TITLE
Error same date/time

### PR DESCRIPTION
**Pull request by adeelahmed057**

### CODE REVIEW

File: app/Http/Controllers/TaskController.php [hunk 1]
```diff
@@ -42,15 +42,21 @@ public function store(Request $request)
...
             'name' => 'required|max:12|regex:/^[\p{L}\s-]+$/u',
+            'date' => 'required',
+            'time' => 'required|unique:tasks'
 
         ]);
 
         $validatedData = $request->validate([
             'name' => ['required', 'max:12','regex:/^[\p{L}\s-]+$/u'],
+            'date' =>['required'],
+            'time' =>['required'],
```
Comment (amanank) [inline app/Http/Controllers/TaskController.php]: Why are you running validation twice?

File: database/migrations/2022_06_18_113401_tasks_table.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('tasks', function (Blueprint $table) {
+            $table->string('date');
+            $table->string('time');
+        });
```
Comment (amanank) [inline database/migrations/2022_06_18_113401_tasks_table.php]: Unique constraint is missing

File: database/migrations/2022_06_18_113401_tasks_table.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::dropIfExists('tasks');
```
Comment (amanank) [inline database/migrations/2022_06_18_113401_tasks_table.php]: this should only drop the two new columns not the entire table

File: resources/views/tasks.blade.php [hunk 4]
```diff
@@ -38,6 +38,28 @@
...
+                </div>
+                </div> 
+            </div>
+
+            <!--------------------Time---------------------------->
+
+            <div class="form-group date" time-provide="timepicker">
+                <label for="task" class="col-sm-3 control-label ">Time</label>
+                    <div class="col-sm-6">
+                        <input type = 'time' name="time" class="form-control" />
```
Comment (amanank) [inline resources/views/tasks.blade.php]: why is 'time' in single quotes and also we only need AM, PM, please read the task description

**Review by amanank — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/TaskController.php (hunk 1 above)
Comment (adeelahmed057) [reply app/Http/Controllers/TaskController.php]: remove unused validation

File: database/migrations/2022_06_18_113401_tasks_table.php (hunk 2 above)
Comment (adeelahmed057) [reply database/migrations/2022_06_18_113401_tasks_table.php]: add unique in both

File: database/migrations/2022_06_18_113401_tasks_table.php (hunk 3 above)
Comment (adeelahmed057) [reply database/migrations/2022_06_18_113401_tasks_table.php]: drop the  new columns.....done

File: database/migrations/2022_06_18_113401_tasks_table.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::dropIfExists('date');
```
Comment (amanank) [inline database/migrations/2022_06_18_113401_tasks_table.php]: You need to drop the constraint first; have you tested this with rollback command?

```sh
php artisan migrate:rollback --step=1
```

Comment (adeelahmed057) [reply database/migrations/2022_06_18_113401_tasks_table.php]: php artisan migrate:refresh command was used...

File: resources/views/tasks.blade.php [hunk 6]
```diff
@@ -58,7 +57,7 @@
             <div class="form-group">
                 <label for="time">Time</label>
                 <select class="form-control {{ $errors->first('time') ? 'is-invalid' : '' }}" name="time">
-                    <option selected>Please Select Time</option>>
+                    <option value="" selected>Please Select Time</option>>
```
Comment (amanank) [inline resources/views/tasks.blade.php]: Remove selected from this option

Comment (adeelahmed057) [reply resources/views/tasks.blade.php]: I have removed selected form this option

**Review by amanank — APPROVED** [review]

**Pull request merged**